### ISSUE TITLE
`required` is ignored in strict mode

### ISSUE THREAD

**Issue by bluemango** [thoughtbot/json-matchers#9]
Because the current version matches everything in strict mode, you aren't able to add optional properties and the `required` key is basically ignored.

This JSON spec will fail if anything but the `id` is present:

    {
      "type": "object",
      "required": [
        "id"
      ],
      "properties": {
        "id": { "type": "number" },
        "content_blocks": {"type": "array"},
        "is_step": {"type": "boolean"},
        "title": {"type": "string"},
        "instructions": {"type": ["string", "null"]}
      },
      "additionalProperties": false
    }

But if I call `JSON::Validator.validate!(path, response.body, strict: false)` then it will pass. 

It would be good to either check for the `required` key and then pass in `strict: false` or give some way to define whether you should use strict mode or not.

See here: https://github.com/ruby-json-schema/json-schema/issues/139

**Comment by seanpdoyle** [comment]
Thanks for the bug report Greg!

I'll make sure to look into this.

**Comment by seanpdoyle** [comment]
@bluemango I can't seem to reproduce this.

I've added a test [on a separate branch](https://github.com/thoughtbot/json-matchers/compare/sd-accept-options), but unfortunately, it's not failing.

Could you add some more context around your bug?

**Comment by seanpdoyle** [comment]
@bluemango scratch that, I spoke too soon.

When I pulled in the JSON schema you provided in your issue, I was able to get a failing test.

In the short term, it might make sense to pass `strict: false` to the matchers that give you grief.

https://github.com/thoughtbot/json-matchers/pull/11

In the long term, we might be able to be smarter about that, and detect the strictness based on the body of the Schema.
